sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala

```diff
@@ -182,8 +181,7 @@ private[columnar] case object RunLengthEncoding extends CompressionScheme {
     private var _uncompressedSize = 0
     private var _compressedSize = 0
 
-    // Using `MutableRow` to store the last value to avoid boxing/unboxing cost.
```

Comment:
do you mean the overhead of boxing is smaller than the overhead of indirections introduced by `SpecificInternalRow`?

Reply:
The benchmark result does. Please let us know if we need other validation, @cloud-fan .

Reply:
I'm fine with it, but it's better to spell out this key finding in the PR description, so that people don't need to infer it from the benchmark results by themselves.

Reply:
Yea, I also think this finding looks interesting. I'm not sure that this rule (`the overhead of boxing is smaller than the overhead of indirections`) always hold on any JVM implementations, but I think its importatnt to notice this performance property for future developement.

Reply:
Thanks for the comments @cloud-fan @dongjoon-hyun and @maropu , and yes my bad for not putting enough info in the PR description. The comment is pretty old and I'm not sure what has changed over the years. By examine flamegraph from a test run I observed boxing **both before and after the change**:

Before:

<img width="1196" alt="Screen Shot 2020-09-14 at 10 30 05 AM" src="https://user-images.githubusercontent.com/506679/93118585-5cfe5600-f675-11ea-916d-213260709b4a.png">

After:

<img width="1191" alt="Screen Shot 2020-09-14 at 10 37 05 AM" src="https://user-images.githubusercontent.com/506679/93119208-43114300-f676-11ea-875c-df497bacd520.png">

Even though the time spent on the boxing part is noticeably less after the change (it could have sth to do with my sample size though).

Perhaps we can open a JIRA to investigate this further? we could have more room for improvement if we can completely eliminate the boxing/unboxing.

Reply:
@sunchao It is hard to guess the difference of boxing between two flamegraphs, can you show the percentage of boxing in each flamegraph?

Reply:
@viirya Sure. Here's complete graphs

Before:

<img width="1194" alt="Screen Shot 2020-09-14 at 11 45 43 AM" src="https://user-images.githubusercontent.com/506679/93125938-a86a3180-f680-11ea-9d31-f48dd804307c.png">

After:

<img width="1193" alt="Screen Shot 2020-09-14 at 11 45 52 AM" src="https://user-images.githubusercontent.com/506679/93125954-abfdb880-f680-11ea-9259-862683db117c.png">

Reply:
Ah sorry @viirya just noticed you were asking for percentage. I calculated. Before the change it was 2.76% and after it is 1.31%.

Reply:
This is nice :-)